serenata_toolbox/chamber_of_deputies/official_missions_dataset.py

```diff
@@ -0,0 +1,131 @@
+import os
+import re
+import urllib
+
+from datetime import date, timedelta
+
+import pandas as pd
+import numpy as np
+
+from bs4 import BeautifulSoup
+
+from serenata_toolbox.datasets.helpers import (
+    save_to_csv,
+    translate_column,
+)
+
+
+class OfficialMissionsDataset:
+
+    URL = (
+        'http://www.camara.leg.br/missao-oficial/missao-pesquisa?'
+        'deputado=1&'
+        'nome-deputado=&'
+        'nome-servidor=&'
+        'dati={0}&'
+        'datf={1}&'
+        'nome-evento='
+    )
+
+    def fetch(self, start_date, end_date):
+        """
+        Fetches official missions within the given date range
+        """
+
+        records = []
+        for r in self.__generate_ranges(start_date, end_date):
+            if os.environ.get('DEBUG') == '1':
+                print(r)
+            for record in self.__fetch_missions_for_range(r[0], r[1]):
+                records.append(record)
```

Comment:
What about a generator to handle that task? I don't know how big this `records` list can get, but memory-wise it worth it to adopt a generator here IMHO (specially if combined with `@property` decorator, so you can access it as `self.records`). I would add `start_date` and `end_date` as `__init__` arguments and save them for the class instance:

```python
class OfficialMissionsDataset:

    …

    def __init__(self, start_date, end_date):
        self.start_date, self.end_date = start_date, end_date

    …

    @property
    def records(self):
        for range_ in self.__generate_ranges():  # I omitted start and end date because you can read them directly in the method
            if os.environ.get('DEBUG') == '1':
                print(range_)
            yield from self.__fetch_missions_for_range(*range_)
```